htdocs/mrp/class/mo.class.php

```diff
@@ -1861,6 +1861,10 @@ public function create(User $user, $notrigger = false)
 			$this->error = 'BadValueForQty';
 			return -1;
 		}
+		if (empty($this->fk_stock_movement) && $this->fields['disable_stock_change']['enabled']) {
```

Comment:
Don't understand this addition.
mrp_production is the table that contains lines for product to consume, product consumed, product to produce and product produced.
We may not know the stock we will use before consumption or production is used. So we $this->fk_stock_movement may be empty and we should be able to insert line in mrp_production to prepare the MO.